From: Sophie Brun <sophie@offensive-security.com>
Date: Tue, 28 Mar 2023 10:29:02 +0200
Subject: Fix mssql scripts

Fix the following issues:
- GetTargetInstances returns only one value.
- #output will always return 0 when scanning a single instance. Add a
  real count of values
- typo in scripts/ms-sql-tables.nse (ouptut != output)

Bug: https://github.com/nmap/nmap/issues/2622
Origin: https://github.com/nmap/nmap/issues/2622
---
 nselib/mssql.lua          | 9 ++++++---
 scripts/ms-sql-tables.nse | 2 +-
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/nselib/mssql.lua b/nselib/mssql.lua
index e275e54..cdf287e 100644
--- a/nselib/mssql.lua
+++ b/nselib/mssql.lua
@@ -3203,8 +3203,9 @@ Helper =
     Helper.Discover( host )
 
     if ( port ) then
-      local status, instances = Helper.GetDiscoveredInstances(host, port)
-      if status then
+      local instances = Helper.GetDiscoveredInstances(host, port)
+
+      if instances then
         return true, instances
       else
         return false, "No SQL Server instance detected on this port"
@@ -3331,10 +3332,12 @@ Helper =
         return nil
       end
       local output = {}
+      local count = 0
       for _, instance in ipairs(instances) do
         output[instance:GetName()] = process_instance(instance)
+	count = count + 1
       end
-      if #output > 0 then
+      if count > 0 then
         return outlib.sorted_by_key(output)
       end
       return nil
diff --git a/scripts/ms-sql-tables.nse b/scripts/ms-sql-tables.nse
index caf0a82..45f32c5 100644
--- a/scripts/ms-sql-tables.nse
+++ b/scripts/ms-sql-tables.nse
@@ -245,7 +245,7 @@ local function process_instance( instance )
   instanceOutput["name"] = string.format( "[%s]", instance:GetName() )
   table.insert( instanceOutput, output )
 
-  return stdnse.format_ouptut(true, instanceOutput)
+  return stdnse.format_output(true, instanceOutput)
 
 end
 
